readme.md

```diff
@@ -1,124 +1,113 @@
 # rehype-react
 
-[![Build Status](https://travis-ci.org/rhysd/rehype-react.svg?branch=master)](https://travis-ci.org/rhysd/rehype-react)
+[![Build Status][travis-badge]][travis-status]
 
-**rehype-react** compiles HAST (HTML Abstract Syntax Tree) to React.  Built on [**remark**](https://github.com/wooorm/remark),
-an extensively tested and pluggable parser.
+Compiles [HAST][] to [React][] with [**rehype**][rehype].
 
-**Why?** Using innerHTML and [dangerouslySetInnerHTML](https://facebook.github.io/react/tips/dangerously-set-inner-html.html) in
-[React.js](http://facebook.github.io/react/) is a common cause of [XSS](https://en.wikipedia.org/wiki/Cross-site_scripting)
-attacks: user input can include script tags and other kinds of active
-content that reaches across domains and harms security. remark-react
-builds a DOM in React, using [React.createElement](https://facebook.github.io/react/docs/top-level-api.html):
-this means that you can display parsed & formatted Markdown content
-in an application without using `dangerouslySetInnerHTML`.
+## Install
 
-## Installation
-
-[npm](https://docs.npmjs.com/cli/install):
+[npm][]:
 
 ```bash
 npm install rehype-react
 ```
 
-## Table of Contents
-
-*   [Programmatic](#programmatic)
-
-    *   [remark.use(react, options)](#remarkusereact-options)
-
-*   [Configuration](#configuration)
-
-*   [Integrations](#integrations)
-
-*   [License](#license)
-
-## Programmatic
-
-### [remark](https://github.com/wooorm/remark#api).[use](https://github.com/wooorm/remark#remarkuseplugin-options)(react, [options](#configuration))
+## Use
 
-**Parameters**
-
-*   `react` — This plugin;
-*   `options` (`Object?`) — See [below](#configuration).
-
-Let’s say `example.js` looks as follows:
+The following example shows how to create a markdown input textarea,
+and corresponding rendered HTML output.  The markdown is processed
+to add a Table of Contents and to render GitHub mentions (and other
+cool GH features), and to highlight code blocks.
 
 ```js
-var React = require('react'),
-    rehype = require('rehype'),
-    reactRenderer = require('rehype-react');
+var react = require('react');
+var unified = require('unified');
+var markdown = require('remark-parse');
+var toc = require('remark-toc');
+var github = require('remark-github');
+var remark2rehype = require('remark-rehype');
+var highlight = require('rehype-highlight');
+var rehype2react = require('rehype-react');
+
+var processor = unified()
+  .use(markdown)
+  .use(toc)
+  .use(github)
+  .use(remark2rehype)
+  .use(highlight)
+  .use(rehype2react);
```

Comment:
React runtime escapes DOM but does not sanitize.  So, it can't help to prevent malicious Markdown document input.

For example, 

```markdown
## Test

<script>
  // malicious code 
</script>
```

Especially, in the environment where not protected by sandbox like Electron, it's important to sanitize some tags and properties. Anyone can access to everything via Markdown.

```markdown
## Malicious Markdown

<script>
// Rendering this markdown document destroys my home directory
require('child_process').spawn('rm', ['-r', '-f', process.env.HOME]);
</script>
```